src/components/Personal-Zone/Searcher/AddResumeDialog/ProgressBar/ProgressBar.tsx

```diff
@@ -59,34 +60,68 @@ const ProgressBar: React.FC<ProgressBarProps> = (props): JSX.Element => {
                     setPreviousJobs={setPreviousJobs}
                     jobName={jobName}
                     setJobName={setJobName} />;
-            case 2:
+            case ResumeStep.NEXT_JOB:
                 return <NextJob
                     roles={allSelectOptions?.roleOptions || []}
                     nextRoles={nextRoles}
                     setNextRoles={setNextRoles}
                     aboutMe={aboutMe}
-                    setAboutMe={setAboutMe} />;
+                    setAboutMe={setAboutMe}/>;
             default:
-                return 'Unknown step';
+                return <></>;
         }
     }
 
-    const handleNext = () => {
-        setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    const handleNext = (): void => {
+        if (validateStep()) {
+            if (activeStep + 1 < steps.length) {
+                setActiveStep((prevActiveStep) => prevActiveStep + 1);
+            } else {
+                submitResume();
+            }
+        }
     };
 
-    const handleBack = () => {
+    const handleBack = (): void => {
         setActiveStep((prevActiveStep) => prevActiveStep - 1);
     };
 
+    const validateStep = (): boolean => {
+        switch (activeStep) {
+            case ResumeStep.PERSONAL_DETAILS: {
+                const alteredSteps: StepInfo[] = steps.slice();
+                alteredSteps[ResumeStep.PERSONAL_DETAILS].errors = {};
+
+                if (!(/^\d{4,10}$/.test(phoneNumber))) {
+                    if (phoneNumber === '') {
+                        alteredSteps[ResumeStep.PERSONAL_DETAILS].errors.phoneError = 'מספר טלפון לא יכול להיות ריק';
```

Comment:
Suggestion: I would do something like the following:

`const phoneErrorMessage = phoneNumber === ''`
   ? `<OPTION_A>`
   : `<OPTION_B>`

And then:
`alteredSteps[ResumeStep.PERSONAL_DETAILS].errors.phoneError` = phoneErrorMessage